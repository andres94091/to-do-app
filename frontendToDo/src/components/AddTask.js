import React, {useEffect, useRef, useState} from 'react'
import axios from 'axios'

export const AddTask = ({setTodos, user, setBandReload}) => {

    const [inputValue, setInputValue] = useState('')
    const [saveTask, setSaveTask] = useState(true)

    const isFirstRun = useRef(true);
    
    async function fetchData() {
        try {
            await axios.post(`http://localhost:5000/api/tasks`, {
                user_id: user,
                task: inputValue
            })
            setInputValue('')
            setBandReload(band => !band)
        } catch (e){
            console.log(e);
            setBandReload(band => !band)
        }
    }
    useEffect(() => {
        if (isFirstRun.current) {
            isFirstRun.current = false;
            return;
        }
        console.log('entre aqui');
        fetchData()
    }, [saveTask])

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if (!inputValue) return
        setTodos(todos => {
            const newTask =  [...todos, {task: inputValue, status: false}]
            return newTask.sort((x, y) => (x.status === y.status) ? 0 : x.status ? 1 : -1)
        })
        setSaveTask(band => !band)
    }

    return (
        <form onSubmit = {handleSubmit}>
            <input
                type= "text"
                value = {inputValue}
                onChange = {handleInputChange}
            />
        </form>
    )
}

export default AddTask
