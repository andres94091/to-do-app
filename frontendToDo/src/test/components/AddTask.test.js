import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios'
import AddTask from '../../components/AddTask';

const setTodos = jest.fn()
const setBandReload = jest.fn()

jest.mock(('axios'), () => ({
  post: jest.fn()
}))

describe('tests in <AddTask/>', () => {

  const wrapper = shallow(<AddTask
    setTodos = {setTodos}
    user = {1}
    setBandReload = {setBandReload}
  />)

  it('should render the component correctly', () => {

    expect(wrapper).toMatchSnapshot();
    
  })

  it('should post the info and clean input', () => {
    const input  = wrapper.find('input')
    const value = 'testing'
    input.simulate('change', {target:{value}})

    wrapper.find('form').simulate('submit', {preventDefault(){}})

    expect(setTodos).toHaveBeenCalled()
    expect(setTodos).toHaveBeenCalledTimes(1)
    expect(axios.post).toHaveBeenCalled()

    const inputValue = input.prop('value')
    expect(inputValue).toBe('')
  })


})
