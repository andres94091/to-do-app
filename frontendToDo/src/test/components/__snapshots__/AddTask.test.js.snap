// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <AddTask/> should render the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
